netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+        private static string GetTargetPath(ITaskItem item)
+        {
+            // first use TargetPath, then Path, then fallback to filename+extension alone
+            foreach (var metadata in s_targetPathMetadata)
+            {
+                var value = item.GetMetadata(metadata);
+
+                if (!String.IsNullOrWhiteSpace(value))
+                {
+                    return value.Replace('\\', '/');
+                }
+            }
+
+            var sourcePath = GetSourcePath(item);
+
+            return Path.GetFileName(sourcePath);
+        }
+
+        private ITaskItem HandleConflict(ITaskItem item1, ITaskItem item2)
+        {
+            var conflictMessage = $"Encountered conflict between {item1.ItemSpec} and {item2.ItemSpec}.";
+
+            var sourcePath1 = GetSourcePath(item1);
+            var sourcePath2 = GetSourcePath(item2);
+
+            var exists1 = File.Exists(sourcePath1);
+            var exists2 = File.Exists(sourcePath2);
+
+            if (!exists1 || !exists2)
+            {
+                var fileMessage = !exists1 ?
+                                    !exists2 ? 
+                                      "both files do" :
+                                      $"{item1.ItemSpec} does" :
+                                  $"{item2.ItemSpec} does";
+
+                Log.LogMessage($"{conflictMessage}.  Could not determine winner because {fileMessage} not exist.");
+                return null;
+            }
+            
+            var assemblyVersion1 = TryGetAssemblyVersion(sourcePath1);
+            var assemblyVersion2 = TryGetAssemblyVersion(sourcePath2);
+
+            // if only one is missing version
+            if (assemblyVersion1 == null ^ assemblyVersion2 == null)
+            {
+                var nonAssembly = assemblyVersion1 == null ? item1.ItemSpec : item2.ItemSpec;
+                Log.LogMessage($"{conflictMessage}. Could not determine a winner because {nonAssembly} is not an assembly.");
+                return null;
+            }
+
+            if (assemblyVersion1 != assemblyVersion2)
+            {
+                if (assemblyVersion1 > assemblyVersion2)
+                {
+                    Log.LogMessage($"{conflictMessage}.  Choosing {item1.ItemSpec} because AssemblyVersion {assemblyVersion1} is greater than {assemblyVersion2}.");
+                    return item1;
+                }
+
+                if (assemblyVersion2 > assemblyVersion1)
+                {
+                    Log.LogMessage($"{conflictMessage}.  Choosing {item2.ItemSpec} because AssemblyVersion {assemblyVersion2} is greater than {assemblyVersion1}.");
+                    return item2;
+                }
+            }
+
+            var fileVersion1 = GetFileVersion(sourcePath1);
+            var fileVersion2 = GetFileVersion(sourcePath2);
+
+            // if only one is missing version
+            if (fileVersion1 == null ^ fileVersion2 == null)
```

Comment:
Then this will be false, and the following condition will be false so we'll skip the comparison.